### PR TITLE
Updated the MicroK8s installation script with kubectl config, flux config

### PR DESCRIPTION
**Pull request by brijrajsingh**
Updated the MicroK8s installation script with kubectl config, and Flux installations and configuration

### CODE REVIEW

**Comment by brijrajsingh** [comment]
@brauliomsft  Please review this PR

File: configmicrok8s.sh [hunk 1]
```diff
@@ -1,16 +1,211 @@
...
+printf '\n Enablings DNS addon for MicroK8s.....\n'
+sudo microk8s enable dns
+sleep 5
+kubectl wait --for=condition=containersReady pod -l k8s-app=kube-dns -n kube-system
+printf '\n microk8s dns enabled successfully ✅\n'
+
+printf "Installing flux 🚧 \n"
+# Install Flux
+curl -s https://fluxcd.io/install.sh | sudo bash
+. <(flux completion bash)
```
Comment (bartr) [inline configmicrok8s.sh:108]: you should echo this to .bashrc so it's always available

File: configmicrok8s.sh [hunk 2]
```diff
@@ -1,16 +1,211 @@
...
+
+# Setup flux
+rm -rf $HOME/$GITOPS_REPO
```
Comment (bartr) [inline configmicrok8s.sh:111]: PiB uses $HOME/gitops for the directory so that path / files can be found

File: configmicrok8s.sh [hunk 3]
```diff
@@ -1,16 +1,211 @@
...
+git clone https://$GITOPS_PAT@github.com/$GITOPS_REPO $HOME/$GITOPS_REPO
+
+cd $HOME/$GITOPS_REPO
+
```
Comment (bartr) [inline configmicrok8s.sh:115]: check for gitops_branch == main

File: configmicrok8s.sh [hunk 4]
```diff
@@ -1,16 +1,211 @@
...
+  annotations:
+    kubernetes.io/service-account.name: admin-user
+type: kubernetes.io/service-account-token
+EOF
+
+TOKEN=$(kubectl get secret admin-user -o jsonpath='{$.data.token}' | base64 -d | sed $'s/$/\\\n/g')
+
+printf "\n ####### Token to connect to Azure ARC starts here ######## \n"
+printf $TOKEN
+printf "\n ####### Token to connect to Azure ARC ends here   ######### \n"
```
Comment (bartr) [inline configmicrok8s.sh:197]: is it necessary to write the token to disk in clear text?

If so, put in the .ssh directory and chmod to min permissions.

**Pull request merged**